### PR TITLE
Support pip 6.x, update dependency versions for tests

### PR DESCRIPTION
**Pull request by jmbowman**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -8,12 +9,13 @@
 root_dir = os.path.abspath(os.path.dirname(__file__))
 requirements_path = os.path.join(root_dir, 'requirements', 'base.txt')
 
-finder = PackageFinder([], [])
-requirements = parse_requirements(requirements_path, finder)
+session = PipSession()
+finder = PackageFinder([], [], session=session)
+requirements = parse_requirements(requirements_path, finder, session=session)
```
Comment (chrisguidry) [inline setup.py]: Whoa, this is cool.

File: requirements/tests.txt [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
+Django==1.6.10
+django-nose==1.3
+requests==2.5.1
+selenium==2.44.0
 
 # Direct dependencies for running tests
 
 # For managing test environments
-tox==1.7.0
+tox==1.8.1
```
Comment (chrisguidry) [inline requirements/tests.txt]: So are all of the version upgrades in this PR related to the pip 6+ upgrade?  Or are some of them connected to the merge from our friend?

**Comment by chrisguidry** [comment]
:+1: Looks good.

File: requirements/tests.txt (hunk 2 above)
Comment (jmbowman) [inline requirements/tests.txt]: None of these upgrades are needed just to support pip 6.x, although the pip/setuptools/virtualenv combination is needed to upgrade any one of those and have them consistent across tox environments (virtualenv includes particular versions of them)

**Pull request merged**